File: custom-bots/node-gateway/src/httpClient.ts
import fetch, { Response } from "node-fetch";

export interface HttpClient {
  post(
    url: string,
    body: unknown,
    headers?: Record<string, string>,
  ): Promise<Response>;
}

export function createHttpClient(): HttpClient {
  return {
    async post(
      url: string,
      body: unknown,
      headers?: Record<string, string>,
    ): Promise<Response> {
      return fetch(url, {
        method: "POST",
        headers: {
          "content-type": "application/json",
          ...(headers ?? {}),
        },
        body: JSON.stringify(body),
        // Basic timeout to avoid hanging connections
        // AbortSignal.timeout is available in Node.js 18+
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        signal: (AbortSignal as any).timeout
          ? // eslint-disable-next-line @typescript-eslint/no-explicit-any
            (AbortSignal as any).timeout(5000)
          : undefined,
      });
    },
  };
}

File: custom-bots/biometric-proxy/app/http_client.py
from __future__ import annotations

from typing import Any

import httpx


class AsyncHttpClient:
    def __init__(self) -> None:
        self._client = httpx.AsyncClient(timeout=10.0)

    async def post(
        self,
        url: str,
        json: dict[str, Any],
        headers: dict[str, str] | None = None,
    ) -> httpx.Response:
        return await self._client.post(url, json=json, headers=headers)

    async def aclose(self) -> None:
        await self._client.aclose()
